### DISCUSSION TITLE
Tests stuck in Started state, never finsh

### DISCUSSION THREAD

**Discussion by borj** [alexjustesen/speedtest-tracker#1841] [Q&A] [answered]
Since 3-4 days ago (since updating to 0.23.1 I believe) I have no successful test runs, every single test is listed as started, and no results. DB connectivity seems to be OK, speedtest inside the container runs OK. Any ideas what/where to look?
![2024-11-26 13_16_11-Window](https://github.com/user-attachments/assets/4a279708-ed2a-443e-bddd-0406ae970eba)
![2024-11-26 13_16_00-Window](https://github.com/user-attachments/assets/dd44086c-4c19-4360-b1b4-81c6abe11f74)
![2024-11-26 13_15_43-Window](https://github.com/user-attachments/assets/b6b841b3-a893-470e-a779-29d05243cfb5)
![2024-11-26 13_15_32-Window](https://github.com/user-attachments/assets/347d6cbf-a656-408f-8a55-6fae6ad56de6)
![2024-11-26 13_15_19-Window](https://github.com/user-attachments/assets/d21e4096-b304-4bd1-bbd1-e114b0eee113)

**Comment by alexjustesen** [comment]
Make sure you have `0.24.x` installed this release fixes stuck states especially when a test fails.

**Reply by borj** [reply]
OK. I have it automated, Watchtower pulls new image when available and the last one it pulled is 0.23.1 Will wait a bit and see if is OK when the newer image arrives.

**Comment by borj** [comment]
Forced an update (now @0.24.1), now it is different but not better. Now the tests fail immediately with "Failed to fetch external IP address, server is likely unable to connect to the internet." The container does have access to the internet (for example if I schedule a test manually it updates the list of servers AND a test from CLI works OK)
![2024-11-26 15_31_43-Window](https://github.com/user-attachments/assets/0573b742-8acb-4327-add4-c76ce9475fc4)
![2024-11-26 15_31_26-Window](https://github.com/user-attachments/assets/e49db207-3e02-4f8a-8dce-be83597adb7d)
![2024-11-26 15_35_53-Window](https://github.com/user-attachments/assets/2649afba-0f15-48c8-852f-a81b9fb5c6d0)

**Reply by svenvg93** [reply, marked as answer]
Are you running any DNS filtering in your network? When a scheduled speedtest is being processed Speedtest Tracker will make a call to http://icanhazip.com/ to get your external IP address. This is done to determine if your server has access to the internet and to check if your external IP address (WAN IP) should be skipped.

https://docs.speedtest-tracker.dev/help/faqs#adguard-and-pi-hole-allow-domains

**Reply by borj** [reply]
I definitely am (Pi-hole), thank you. Will take a look in a minute.

**Reply by borj** [reply]
Thank you very much, whitelisting the domain solved it.

**Comment by borj** [comment]
As pointed by [Sven van Ginkel](https://github.com/svenvg93) - the domain used by speedtest tracker for external IP fetching was filtered by my local DNS filter (Pi-hole) Whitelisting the domain (icanhazip.com) solves the issue.
